gbpservice/nfp/configurator/agents/loadbalancer_v1.py

```diff
@@ -0,0 +1,655 @@
+                'get_logical_device',
+                pool_id=pool_id
+            )
+        )
+
+    def update_status(self, obj_type, obj_id, status):
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_status',
+               'kwargs': [{'obj_type': obj_type,
+                           'obj_id': obj_id,
+                           'status': status}]
+               }
+        self._notification(msg)
+
+    def update_pool_stats(self, pool_id, stats):
+        LOG.info("[LbaaSRpcSender]: Update pool stats called")
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_pool_stats',
+               'kwargs': [{'pool_id': pool_id,
+                           'stats': stats}]
+               }
+        self._notification(msg)
+
+
+"""Implements APIs invoked by configurator for processing RPC messages.
+
+RPC client of configurator module receives RPC messages from REST server
+and invokes the API of this class. The instance of this class is registered
+with configurator module using register_service_agent API. Configurator module
+identifies the service agent object based on service type and invokes ones of
+the methods of this class to configure the device.
+
+"""
+
+
+class LBaaSRpcManager(agent_base.AgentBaseRPCManager):
+    def __init__(self, sc, conf):
+        """Instantiates child and parent class objects.
+
+        Passes the instances of core service controller and oslo configuration
+        to parent instance inorder to provide event enqueue facility for batch
+        processing event.
+
+        :param sc: Service Controller object that is used for interfacing
+        with core service controller.
+        :param conf: Configuration object that is used for configuration
+        parameter access.
+
+        """
+
+        super(LBaaSRpcManager, self).__init__(sc, conf)
+
+    def create_vip(self, context, vip):
+
+        arg_dict = {'context': context,
+                    'vip': vip,
+                    'serialize': True,
+                    'binding_key': vip['pool_id']
+                    }
+        ev = self._sc.new_event(id='CREATE_VIP', data=arg_dict)
+        self._sc.post_event(ev)
+
+    def update_vip(self, context, old_vip, vip):
+        arg_dict = {'context': context,
+                    'old_vip': old_vip,
+                    'vip': vip,
+                    'serialize': True,
+                    'binding_key': vip['pool_id']
+                    }
+        ev = self._sc.new_event(id='UPDATE_VIP', data=arg_dict)
+        self._sc.post_event(ev)
+
+    def delete_vip(self, context, vip):
+
+        arg_dict = {'context': context,
+                    'vip': vip,
+                    'serialize': True,
+                    'binding_key': vip['pool_id']
+                    }
+        ev = self._sc.new_event(id='DELETE_VIP', data=arg_dict)
```

Comment:
You can have a common method for sending event so that new_event and post_event need not be repeated in all the functions.